### PR TITLE
add deployment config for cloudigrade-celery

### PR DESCRIPTION
**Pull request by infinitewarp**
https://github.com/cloudigrade/cloudigrade/issues/243

### CODE REVIEW

**Comment by infinitewarp** [comment]
Demo... pull and run it yourself, or look over my shoulder? Here's a screenshot of how it appears to be running:

File: deployment/ocp/cloudigrade.yml [hunk 1]
```diff
@@ -8,7 +8,123 @@ objects:
...
+    revisionHistoryLimit: 10
+    selector:
+      name: ${NAME}-celery
+    template:
+      metadata:
+        labels:
+          name: ${NAME}-celery
+      spec:
+        containers:
+        - name: ${NAME}-celery
```
Comment (ilyavw) [inline deployment/ocp/cloudigrade.yml]: This container is missing readiness and liveness probes, we need some method for checking if it's healthy.

File: deployment/ocp/cloudigrade.yml [hunk 2]
```diff
@@ -8,7 +8,123 @@ objects:
 - kind: DeploymentConfig
   apiVersion: v1
   metadata:
-    name: cloudigrade
+    name: ${NAME}-celery
+  spec:
```
Comment (ilyavw) [inline deployment/ocp/cloudigrade.yml]: You do not have a a [deployment strategy](https://docs.openshift.com/dedicated/dev_guide/deployments/deployment_strategies.html) defined. Is celery compatible with a rolling deployment strategy (the default)? Or would using recreate here be better?

If you'd like to switch to using a recreate simply add 
```
strategy:
  type: Recreate
```
between the metadata and spec.

File: deployment/ocp/cloudigrade.yml [hunk 3]
```diff
@@ -8,7 +8,123 @@ objects:
...
+    replicas: 1
```
Comment (ilyavw) [inline deployment/ocp/cloudigrade.yml]: Should this be configurable? Does this container support multiple pods running? Should we define [autoscaling](https://docs.openshift.com/dedicated/dev_guide/pod_autoscaling.html)?

File: deployment/ocp/cloudigrade.yml (hunk 2 above)
Comment (infinitewarp) [reply deployment/ocp/cloudigrade.yml]: Does `strategy` belong *in* the `spec`, not as a peer? That's where it lives for the other `DeploymentConfig`s.

Comment (ilyavw) [reply deployment/ocp/cloudigrade.yml]: That's correct, my mistake!

**Comment by infinitewarp** [comment]
@ilyavw I spent some time trying to get autoscaling to work per your review comment, but I time-boxed myself to end-of-today and didn't manage to get it working in that time. I hit a couple of problems along the way, although I'm willing to chock it up to ignorance on my part wrt configuration. 🙂 

- Functional autoscaling seems to require enabling metrics on the cluster, but I don't see how to do this. `oc cluster up` takes an optional `--metrics` flag (which is also marked "experimental"), but that didn't seem to enable what autoscaling needed.
- The necessary object `HorizontalPodAutoscaler` configuration requires `apiVersion: autoscaling/v1` and `apiVersion: extensions/v1beta1`. These seem to be fundamentally incompatible with the `apiVersion: v1` in our `cloudigrade.yml` `Template` definition, and I couldn't find a way to make them play nicely together. I found a workaround that suggested putting the `HorizontalPodAutoscaler` object into its own file to import, and that does seem to work but feels wrong.

Since I didn't get it totally working, I shoved my progress into another branch if we want to look at it or pick it up later: https://github.com/cloudigrade/cloudigrade/tree/243-worker-autoscaling

Also, since we don't have autoscaling on anything else yet but *eventually* probably want it on more things, I figure it's reasonable to punt on it for this PR and just get it working everywhere later.

**Pull request merged**